### PR TITLE
validation point-in-time fix

### PR DESCRIPTION
**Pull request by deep-outcome**
See https://github.com/dotnet/AspNetCore.Docs/issues/33837#issuecomment-2405687724.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/configuration/options.md](https://github.com/dotnet/AspNetCore.Docs/blob/0011040c24e75c42dc455c12b56335d57f705080/aspnetcore/fundamentals/configuration/options.md) | [Options pattern in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?branch=pr-en-us-33844) |

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/options.md [hunk 1]
```diff
@@ -208,7 +208,12 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time `TOption` instance is created. That means for instance when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601.Value) occurs in a request pipeline or when
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:212]:
```suggestion
access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601) occurs in a request pipeline or when 
```

File: aspnetcore/fundamentals/configuration/options.md [hunk 2]
```diff
@@ -208,7 +208,12 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time `TOption` instance is created. That means for instance when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601.Value) occurs in a request pipeline or when 
+[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.IOptionsMonitor%601.Get*) is called on settings present. This implies that
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:213]:
```suggestion
[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.OptionsMonitor`1.Get(System.String)) is called on settings present. This implies that 
```

File: aspnetcore/fundamentals/configuration/options.md [hunk 3]
```diff
@@ -208,7 +208,12 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time `TOption` instance is created. That means for instance when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601.Value) occurs in a request pipeline or when 
+[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.IOptionsMonitor%601.Get*) is called on settings present. This implies that 
+after settings reload validation runs again. ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once created.
+
+To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:216]:
```suggestion
To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart``1(Microsoft.Extensions.Options.OptionsBuilder{``0})>in `Program.cs`:
```

File: aspnetcore/fundamentals/configuration/options.md [hunk 4]
```diff
@@ -208,7 +208,12 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time `TOption` instance is created. That means for instance when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601.Value) occurs in a request pipeline or when 
+[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.IOptionsMonitor%601.Get*) is called on settings present. This implies that 
+after settings reload validation runs again. ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once created.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:214]:
```suggestion
after settings reload validation runs again. The ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once created.
```

**Review by tdykstra — APPROVED** [review]

File: aspnetcore/fundamentals/configuration/options.md [hunk 5]
```diff
@@ -208,7 +208,11 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time a `TOption` instance is created. That means, for instance, when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601) occurs in a request pipeline or when 
+[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.OptionsMonitor`1.Get(System.String)) is called on settings present. After settings reload validation runs again. The ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once they are created.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:213]:
```suggestion
[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.OptionsMonitor`1.Get(System.String)) is called on settings present. After settings are reloaded, validation runs again. The ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once they are created.
```

File: aspnetcore/fundamentals/configuration/options.md [hunk 6]
```diff
@@ -208,7 +208,11 @@ Options validation also supports <xref:System.ComponentModel.DataAnnotations.IVa
 
 ### `ValidateOnStart`
 
-Options validation runs the first time an <xref:Microsoft.Extensions.Options.IOptions%601>, <xref:Microsoft.Extensions.Options.IOptionsSnapshot%601>, or <xref:Microsoft.Extensions.Options.IOptionsMonitor%601> implementation is created. To run options validation eagerly, when the app starts, call <xref:Microsoft.Extensions.DependencyInjection.OptionsBuilderExtensions.ValidateOnStart%2A> in `Program.cs`:
+Options validation runs the first time a `TOption` instance is created. That means, for instance, when first 
+access to [`IOptionsSnapshot<TOptions>.Value`](xref:Microsoft.Extensions.Options.IOptionsSnapshot%601) occurs in a request pipeline or when 
+[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.OptionsMonitor`1.Get(System.String)) is called on settings present. After settings are reloaded, validation runs again. The ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once they are created.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/configuration/options.md:213]:
```suggestion
[`IOptionsMonitor<TOptions>.Get(string)`](xref:Microsoft.Extensions.Options.OptionsMonitor`1.Get(System.String)) is called on settings present. After settings are reloaded, validation runs again. The ASP.NET Core runtime uses <xref:Microsoft.Extensions.Options.OptionsCache%601> to cache the options instance once it is created.
```

**Comment by tdykstra** [comment]
@bravequickcleverfibreyarn The links are fixed, and I made some minor changes to the text -- take a look and if everything looks good to you we can merge the PR.

**Comment by bravequickcleverfibreyarn** [comment]
I see no problem.

**Pull request merged**